pv211_utils/beir/eval.py

```diff
@@ -0,0 +1,34 @@
+from typing import Set, Optional
+
+from pv211_utils.eval import EvaluationBase
+from .entities import BeirJudgementBase
+from .irsystem import BeirIRSystemBase
+
+
+class BeirEvaluation(EvaluationBase):
+    """A Generic information retrieval system evaluation.
+
+    Parameters
+    ----------
+    system : BeirIRSystemBase
+        The information retrieval system.
+    judgements : set of BeirJudgementBase
+        Pairs of queries and relevant documents.
+    k : int, optional
+        Parameter defining evaluation depth. Default is 10.
+    leaderboard : BeirLeaderboard or None, optional
+        A leaderboard to which we may later submit evaluation results.
+        If None, then evaluation results will not be submitted. Default is None.
+    author_name : str or None, optional
+        The name of the author submitted to the leaderboard.
+        If None, then the result will not be submitted. Default is None.
+    num_workers : int or None, optional
+        The number of processes used to compute the mean average precision.
+        If None, all available CPUs will be used. Default is 1.
+    """
+    def __init__(self, system: BeirIRSystemBase, judgements: Set[BeirJudgementBase], k: Optional[int] = 10,
```

Comment:
I do not think a single value of `k` makes sense for all the different BEIR datasets. Let's leave it undefined? The `notebooks/beir_cqadupstack.ipynb` notebook also needs to be updated accordingly.

```suggestion
    k : int
        Parameter defining evaluation depth.
    leaderboard : BeirLeaderboard or None, optional
        A leaderboard to which we may later submit evaluation results.
        If None, then evaluation results will not be submitted. Default is None.
    author_name : str or None, optional
        The name of the author submitted to the leaderboard.
        If None, then the result will not be submitted. Default is None.
    num_workers : int or None, optional
        The number of processes used to compute the mean average precision.
        If None, all available CPUs will be used. Default is 1.
    """
    def __init__(self, system: BeirIRSystemBase, judgements: Set[BeirJudgementBase], k,
```

Reply:
should be done